core/domain/state_domain.py

```diff
@@ -1892,6 +2097,65 @@ def from_dict(cls, recorded_voiceovers_dict):
 
         return cls(voiceovers_mapping)
 
+    def to_proto(self):
+        """Returns a proto representation of the recorded voiceovers object.
+
+        Returns:
+            RecordedVoicovers. The recorded voiceovers proto object.
+        """
+        voiceover_content_mapping_protos = (
+            self._to_voiceovers_content_mapping_protos(
+                self.voiceovers_mapping))
+
+        recorded_voiceover_proto = languages_pb2.RecordedVoiceovers(
+            voiceover_content_mapping=voiceover_content_mapping_protos)
+
+        return recorded_voiceover_proto
+
+    @classmethod
+    def _to_voiceovers_content_mapping_protos(cls, voiceovers_mapping):
+        """Creates a VoiceoverContentMapping proto object.
+
+        Args:
+            voiceovers_mapping: dict. A dict mapping the content Ids
+                to the dicts which is the map of abbreviated code of the
+                languages to the Voiceover objects.
+
+        Returns:
+            RecordedVoiceovers. The recorded voiceovers proto object.
+        """
+        language_to_content_id_voiceover_file_map = (
+            collections.defaultdict(dict))
+        for (content_id, language_code_to_voiceover) in (
+            voiceovers_mapping.items()):
+            for (language_code, voiceover) in (
+                language_code_to_voiceover.items()):
+                voiceover_proto = voiceover.to_proto()
+                language_to_content_id_voiceover_file_map[language_code][content_id] = voiceover_proto # pylint: disable=line-too-long
+
+        language_code_to_enum_map = {
+            'en': languages_pb2.LanguageType.ENGLISH,
+            'hi-en': languages_pb2.LanguageType.HINGLISH,
+            'ar': languages_pb2.LanguageType.ARABIC,
+            'hi': languages_pb2.LanguageType.HINDI
```

Comment:
I'm wondering whether this should be tied to an android_validation_constants array to keep it centrally defined with the other Android configurations. @seanlip do you have any preference on this?

Reply:
Currently, in `android_validation_constants.py` file ->

```
# This is linked to SUPPORTED_CONTENT_LANGUAGES_FOR_ANDROID in constants.ts.
SUPPORTED_LANGUAGES = ['en']
```